### PR TITLE
chore(authenticator): UI fixes

### PR DESCRIPTION
**Pull request by dnys1**
*Issue #, if available:*

*Description of changes:*
- Phone code for all phone number fields
- Show required fields for email + phone number case
- Improvements for dial code list
- Fix social buttons
- Fix confirm sign up/sign in flows
- Remove remember device
- Add exception handler
- Various minor UI fixes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_authenticator/lib/src/utils/country_code.dart [hunk 1]
```diff
@@ -25,6 +25,7 @@ class Country {
 
 /// List of country information
 const countryCodes = [
+  Country('US', '1', CountryResolverKey.us$),
```
Comment (Jordan-Nelson) [inline packages/amplify_authenticator/lib/src/utils/country_code.dart:28]: Would be fine with a separate PR for this, but maybe we order these by the code if we want US/+1 as the first on the list?

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/amplify_authenticator/lib/amplify_authenticator.dart [hunk 2]
```diff
@@ -65,6 +66,12 @@ export 'src/widgets/form_field.dart'
         ConfirmSignInFormField,
         ConfirmSignUpFormField;
 
+/// {@template amplify_authenticator.exception_handler}
+/// A user-specified exception handler for errors originating within the
+/// [Authenticator].
+/// {@endtemplate}
+typedef ExceptionHandler = void Function(AuthenticatorException);
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/amplify_authenticator.dart:73]: 💯

File: packages/amplify_authenticator/lib/src/enums/gender.dart [hunk 3]
```diff
@@ -1,3 +1,18 @@
...
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+*/
+
 enum Gender {
   male,
   female,
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/src/enums/gender.dart:18]: maybe add 
`genderNonbinary`, `preferNotToSay`

File: packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart [hunk 4]
```diff
@@ -1,153 +1,158 @@
...
 
-    final hintText = widget.hintText == null
-        ? inputResolver.resolve(context, widget.hintTextKey!)
-        : widget.hintText!;
+  @override
+  Widget get prefix => Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: InkWell(
+          child: Row(
+            mainAxisSize: MainAxisSize.min,
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart:69]: 👍

File: packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart [hunk 5]
```diff
@@ -1,153 +1,158 @@
...
-                          /// If search string is empty, display full list; otherwise, filter
-                          newValue.isEmpty
-                              ? _countries.value = countryCodes
-                              : _countries.value = _countries.value
-                                  .where((c) => countryResolver
-                                      .resolve(context, c.key)
-                                      .toLowerCase()
-                                      .contains(newValue.toLowerCase()))
-                                  .toList();
+                          _searchVal = searchVal;
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart:122]: When I did the initial implementation I tried to use a TextController for this, but  Flutter would throw an exception about an invalid key (similar to the one we we fixed earlier). Did you see this as well?

File: packages/amplify_authenticator/lib/src/mixins/authenticator_radio_field.dart [hunk 6]
```diff
@@ -1,3 +1,19 @@
+/*
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/src/mixins/authenticator_radio_field.dart:1]: Thanks for catching these missing licenses.

File: packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart (hunk 5 above)
Comment (dnys1) [reply packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart:122]: I can't remember the earlier issue, but I haven't had any problems with this.

File: packages/amplify_authenticator/lib/src/utils/country_code.dart (hunk 1 above)
Comment (haverchuck) [reply packages/amplify_authenticator/lib/src/utils/country_code.dart:28]: This could come as a fast follow, but I was hoping we could use the device locale to attempt to set the default?

File: packages/amplify_authenticator/lib/src/enums/gender.dart (hunk 3 above)
Comment (dnys1) [reply packages/amplify_authenticator/lib/src/enums/gender.dart:18]: Okay, will probably update this later since it's not even being used for anything yet. We just have gender as a text field still.

**Review by haverchuck — APPROVED** [review]
LGTM! We need to reset the viewModel.username when switch is toggled, and then 🚢 .
